src/store/dappChainStore.ts

```diff
@@ -220,50 +206,53 @@ export default {
+        throw new Error("Expected dposUser to be initialized")
+      }
+      const dposUser = await state.dposUser
       const web3js = state.web3
-      const accounts = await web3js.eth.getAccounts()
-      if (accounts.length === 0) return 0
-      const address = accounts[0]
       try {
-        let result = await state.dposUser.ethereumLoom.balanceOf(address)
+        debug("ethereumLoom.balanceOf")
+        let result = await dposUser.ethereumLoom.balanceOf(dposUser.ethAddress)
+        debug("ethereumLoom.balanceOf",result.toString())
         let balance = web3js.utils.fromWei(result.toString())
-        let limitDecimals = parseFloat(balance).toFixed(2)
-        return limitDecimals
+        const mainnetBalance = parseFloat(balance).toFixed(2)
+        const userBalance = rootState.DPOS.userBalance
+        commit("DPOS/setUserBalance",Object.assign(userBalance,{mainnetBalance}),{root:true})
+        return mainnetBalance
       } catch (err) {
         commit('setErrorMsg', {msg: "Error getting metamask balance", forever: false, report:true, cause:err}, {root: true})
         return 0
       }
     },
     async initDposUser({ state, rootState, getters, dispatch, commit }) {
+      console.log("initdpos user")
       if (!rootState.DPOS.web3) {    
         await dispatch("DPOS/initWeb3Local", null, { root: true })
       }
- 
-      const network = state.currentChain.chainId
-      let user 
-      try { 
-        user = await DPOSUser.createEthSignMetamaskUserAsync(		
-          rootState.DPOS.web3,
-          state.currentChain.dappchainEndpoint,
-          network,
-          GW_ADDRESS || GW_ADDRESS || state.currentChain["gatewayAddress"],
-          LOOM_ADDRESS || LoomTokenJSON.networks[network].address
-        )
-
-        reconfigureClient(user._client, commit)
-        commit("setDPOSUser", user)
-        
-      } catch(err) {
-
+      const network = state.networkId
+      // set state dposUser to be a Promise<dposUser> so that components caalling it don't complain or go and try to init another dpos user...
+      const user = DPOSUserV3.createEthSignMetamaskUserAsync({
+        web3: rootState.DPOS.web3,
+        dappchainEndpoint: state.chainUrls[state.networkId].dappchainEndpoint,
+        chainId: network,
+        gatewayAddress: GW_ADDRESS || state.currentChain["gatewayAddress"],
+        version: 1
+      })
+      .then(user => {
+        reconfigureClient(user.client, commit)
+        console.log('dposUser ready')
+        return user
+      })
+      .catch((err) => {
         console.log(err)
         commit('setErrorMsg', {msg: "Error initDposUser", forever: false, report:true, cause:err}, {root: true}) 
-      }
-
-
-
+        return null
+      })
+      // set the promise
+      commit("setDPOSUserV3", user)
     },
     // TODO: this is added to fix mismatched account mapping issues, remove once all users are fixed.
     async switchDposUser({ state, rootState, getters, dispatch, commit }, payload) {
```

Comment:
Is this called anywhere useful?

Reply:
here https://github.com/loomnetwork/dashboard/blob/dpos-v3/src/views/MobileAccount.vue#L407